tools/autograd/templates/Functions.cpp

```diff
@@ -108,6 +108,24 @@ Tensor norm_backward(Tensor grad, const Tensor & self, const Scalar & p_, Tensor
   return norm_backward(grad, self, p_, norm);
 }
 
+Tensor pow_backward(Tensor grad, const Tensor & self, const Scalar & exponent_) {
+  double exponent = exponent_.toDouble();
+  if (exponent == 0.0) {
+    return zeros_like(self);
+  } else {
+    return grad * exponent * self.pow(exponent - 1);
+  }
+}
+
+Tensor pow_backward_self(Tensor grad, const Tensor & self, const Tensor & exponent) {
+  Tensor zero_mask = (exponent == 0.0);
+  return at::where(zero_mask, zeros_like(self), grad * exponent * self.pow(exponent - 1));
```

Comment:
This line and the one above makes two allocations of the size of the input tensor. Are there any perf implications?

Reply:
Thanks for catching this. There can be one fix:
```
return at::where((exponent == 0.0), at::zeros({1}, grad.type()), grad * exponent * self.pow(exponent - 1));
```

I don't think